modules/storages/app/common/storages/peripherals/storage_interaction/authentication_strategies/nextcloud_strategies.rb

```diff
@@ -37,10 +37,24 @@ module NextcloudStrategies
             ::Storages::Peripherals::StorageInteraction::AuthenticationStrategies::BasicAuth.strategy
           end
 
-          UserBound = ->(user:) do
-            ::Storages::Peripherals::StorageInteraction::AuthenticationStrategies::OAuthUserToken
-              .strategy
-              .with_user(user)
+          UserBound = ->(user:, storage:) do
```

Comment:
Should we, instead of passing the whole storage, just sending its audience?

Reply:
@mereghost I think sending just the audience is something very specific. We might fall into the same pit we had for requests, that were build for specific nextcloud and then we had trouble in writing those for OneDrive.

My thoughts are coming from a yesterdays experiment, were Jan and me created an Azure SSO, and we were able  to make server to server request with fully obfuscated audiences, it was a weird UUID.

So, point is, maybe we should have a better abstraction here. What is the entity, that holds the information about all configured authentication methods? right now, it is the storage object.

Reply:
I'd +1 what Eric said.

Also note that even accepting a parameter like `sso_supported` will also spread knowledge, because the caller of this thing, will suddenly have to _know_ about SSO, to pick an auth strategy, when previously it just needed to know that an auth strategy is broadly related to a user and a storage.

Reply:
> true, but not here. this adapter is called nextcloud_strategies. it is bound to a storage type.

I'd say that it has to know about the "authentication bits" of nextcloud, not the storage itself.

I get the feeling that we are missing an actual abstraction here. We started to draft this with the `OAuthConfiguration` but maybe that was the wrong name.